src/neptune/new/handler.py

```diff
@@ -156,12 +166,14 @@ def assign(self, value, wait: bool = False) -> None:
         """
         with self._container.lock():
             attr = self._container.get_attribute(self._path)
-            if attr:
-                if isinstance(value, Handler):
-                    value = ValueCopy(value)
-                attr.process_assignment(value, wait)
-            else:
-                self._container.define(self._path, value, wait)
+            if not attr:
```

Comment:
> then assign original (before we used casted value for assignment) value to newly created attribute
I had to give up this idea, so in case of creating attribute, we're assigning value casted by general `cast_value` function instead of passing responsibility for casting to attribute.

This will have to be discussed.